### PR TITLE
adds simple first views on model callbacks

### PR DESCRIPTION
**Pull request by jhoetter**

### CODE REVIEW

**Comment by jhoetter** [comment]
Run this together with:
- [Proxy PR](https://github.com/code-kern-ai/refinery-gateway-proxy/pull/6)
- [Gateway PR](https://github.com/code-kern-ai/refinery-gateway/pull/39)
- [Submodule PR](https://github.com/code-kern-ai/refinery-submodule-model/pull/7)

You can add a model callback using the Python SDK in branch `external-information-sources` ([PR](https://github.com/code-kern-ai/refinery-python-sdk/pull/22)) as follows:
```python
from refinery import Client

user_name = "your-mail" # this is the email you log in with
password = "your-pw"
project_id = "your-clickbait-sample-project-id" # can be found in the URL of the web application

client = Client(user_name, password, project_id, uri="http://localhost:4455")

from sklearn.linear_model import LogisticRegression
clf = LogisticRegression()

from refinery.adapter.sklearn import build_classification_dataset
data = build_classification_dataset(client, "headline", "__clickbait", "distilbert-base-uncased", num_train=50)
clf.fit(data["train"]["inputs"], data["train"]["labels"])

from refinery.callbacks.sklearn import SklearnCallback
callback = SklearnCallback(
    client, 
    clf,
    "clickbait", 
)

callback.run(data["train"]["inputs"], data["train"]["index"])
callback.run(data["test"]["inputs"], data["test"]["index"])
```

File: src/app/app.module.ts [hunk 1]
```diff
@@ -23,6 +23,8 @@ import { IntercomModule } from 'ng-intercom';
 import { NotificationCenterModule } from './notification-center/notification-center.module';
 import { RecordIDEModule } from './record-ide/record-ide.module';
 import { ModelDownloadModule } from './model-download/model-download.module';
+import { ModelCallbackComponent } from './model-callbacks/components/model-callbacks.component';
```
Comment (JWittmeyer) [inline src/app/app.module.ts]: not nessesarry imo

File: src/app/model-callbacks/components/model-callbacks.component.html [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                                    <path fill-rule="evenodd"
+                                        d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
+                                        clip-rule="evenodd" />
+                                </svg>
+                            </button>
+                        </div>
+                    </ng-template>
+
+                    <div class="flex justify-center overflow-visible">
+                        <button [routerLink]="['../heuristics']" data-tip="Go to the model callbacks"
```
Comment (JWittmeyer) [inline src/app/model-callbacks/components/model-callbacks.component.html]: data-tip doesnt macht route

File: src/app/model-callbacks/components/model-callbacks.component.html [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+                                        <!-- <div class="text-green-800 float-right cursor-pointer"
+                                            [routerLink]="[source.informationSourceType == 'ZERO_SHOT'?('../zero-shot/' + source.id):('./' + source.id)]"
+                                            [queryParams]="{type: source.informationSourceType, status: source?.state}">
+                                            Details
+                                            <svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6 inline-block"
+                                                fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
+                                                <path stroke-linecap="round" stroke-linejoin="round"
+                                                    d="M13 7l5 5m0 0l-5 5m5-5H6" />
+                                            </svg>
+                                        </div> -->
```
Comment (JWittmeyer) [inline src/app/model-callbacks/components/model-callbacks.component.html]: comment block

File: src/app/model-callbacks/components/model-callbacks.component.ts [hunk 4]
```diff
@@ -0,0 +1,271 @@
+import { Component, OnDestroy, OnInit, ViewChild, ElementRef } from '@angular/core';
```
Comment (JWittmeyer) [inline src/app/model-callbacks/components/model-callbacks.component.ts:1]: seems like there is a lot of unnecessary clutter in the ts

**Comment by JWittmeyer** [comment]
Version overview doesn't seem to work on your branch
image

- [x] resolved

**Comment by JWittmeyer** [comment]
websocket integration seems to be missing for stats - weren't refreshed on the overview page until I refreshed manually)
websocket integration seems to be missing for overview (newly created model callback)

- [x] resolved

**Comment by JWittmeyer** [comment]
no option to change the comment or the name of the information source

- [ ] resolved
- [x] ignored

**Comment by JWittmeyer** [comment]
lookup lists buttons have differing font weight
image

(actions might not be as straightforward since its created with the kern dropdown - making all bald would do the trick imo (also fits with other buttons on heuristic page)

- [x] resolved

**Comment by JWittmeyer** [comment]
Maybe a UX issue we don't have to solve now.

For me the usage/integration feels a bit weird. 
The model callback is somewhat hidden behind the heuristic page, the model callback block in the databrowser clearly depicts it as something else than a heuristic.

Also for clarity/differentiation, it might be a good idea to have a model page with callbacks & weak supervision combined (since weak supervision is basically our model and we want to compare it to a production model). Would also help to differentiate between the gravity of the results and give us the nice stat overview for the weak supervision :) 

Another thing to help me identify it for what it is could be to add a hint in the databrowser or a graying it out if no model callback is present in the project.

- [ ] discussed
- [x] issue created
- [ ] resolved

**Comment by JWittmeyer** [comment]
new weak supervision & model callback labels aren't vertically aligned
image

classes `inline-flex items-center` should do the trick. also padding right seems a bit low with `pr-0.5` (tested with pr removed & `px-1` instead of `px-2`)

image

- [x] resolved

File: src/app/app.module.ts (hunk 1 above)
Comment (jhoetter) [reply src/app/app.module.ts]: true

File: src/app/model-callbacks/components/model-callbacks.component.html (hunk 2 above)
Comment (jhoetter) [reply src/app/model-callbacks/components/model-callbacks.component.html]: true

File: src/app/model-callbacks/components/model-callbacks.component.html (hunk 3 above)
Comment (jhoetter) [reply src/app/model-callbacks/components/model-callbacks.component.html]: removed it :)

File: src/app/model-callbacks/components/model-callbacks.component.ts (hunk 4 above)
Comment (jhoetter) [reply src/app/model-callbacks/components/model-callbacks.component.ts:1]: clutter removed.

**Review by JWittmeyer — APPROVED** [review]
